proposals/infra/INF-0004-binding-resource-binding-diagnostics.md

```diff
@@ -0,0 +1,212 @@
+* Proposal: [0004](0004-binding-prefixes.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: TBD
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+*During the review process, add the following fields as needed:*
+
+* PRs: [#87578](https://github.com/llvm/llvm-project/pull/87578)
+* Issues: [#57886](https://github.com/llvm/llvm-project/issues/57886)
+
+## Introduction
+
+There's a specific set of valid types that can be bound as a resource in HLSL.
+When invalid types are bound to resources, helpful diagnostics need to be emitted
+so that the user knows why the bound type is invalid, or why the binding prefix
+isn't correct.
+
+## Motivation
+
+There are currently several cases in which diagnostics that are related to

```

Comment:
What does "currently" mean in the context of clang?  When I run the example on line 24 in clang I don't get any error message.

I suspect that this is describing the DXC behavior?

Reply:
Yes, it is describing DXC behavior